app/controllers/houses_controller.rb

```diff
@@ -0,0 +1,55 @@
+class HousesController < ApplicationController
+  def index
+    @houses = current_user.houses
+  end
+
+  def new
+    @house = House.new
+  end
+
+  def edit
+    @house = House.find(params[:id])
+  end
+
+  def create
+    @house = House.new(user_params)
+    @house.user = current_user
+    @house.save
+    redirect_to houses_path
+  end
+
+  def show
+    @house = House.find(params[:id])
+  end
+
+  def update
+    @house = House.find(params[:id])
+    if @house.update(user_params)
+      redirect_to house_path(@house.id)
+    end
+end
+
+  def destroy
+    @house = House.find(params[:id])
+    @house.destroy
+    redirect_to houses_path
+  end
+
+  def create_device
```

Comment:
not sure that is working. You didn't defined the route. and you didn't test it :(